src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,135 @@
+    grant_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_e['operationParam']['operationFrequencyRange']\
+                             ['lowFrequency'] = 3650000000
+    grant_e['operationParam']['operationFrequencyRange']\
+                             ['highFrequency'] = 3660000000
+    grant_request = [grant_a, grant_c, grant_e]
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(registration_request,
+                                                           grant_request)
+    # First Heartbeat
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids, [None, None, None])
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # check Heartbeat response 
+    self.assertEqual(len(first_hb_response), len(grant_ids))
+    for response_num, resp in enumerate(first_hb_response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_expire_time = datetime.strptime(resp['grantExpireTime'],\
+                                              '%Y-%m-%dT%H:%M:%SZ')
+        transmit_expire_time = datetime.strptime(resp['transmitExpireTime'],\
+                                                 '%Y-%m-%dT%H:%M:%SZ')
+        self.assertLessEqual((transmit_expire_time - \
+                               datetime.utcnow()).total_seconds(), 240)
+        self.assertLessEqual(transmit_expire_time, grant_expire_time)
+    first_hb_transmit_expire_time = (first_hb_response[0]['transmitExpireTime'],\
+                                   first_hb_response[1]['transmitExpireTime'],\
+                                   first_hb_response[2]['transmitExpireTime'])
+    # STEP 2
+    # load and inject FSS data with Overlapping Frequency of CBSD
+    fss_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record.json')))
+    self._sas_admin.InjectFss({'record': fss_e})   
+    # load and inject GWPZ data with Overlapping Frequency of CBSD 
+    gwpz_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record.json')))  
+    self._sas_admin.InjectWisp({'record': gwpz_c})      
+    # STEP 3
+    # Trigger daily activities and wait for it to get it complete
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+    # Send Heartbeat after triggering the IAP
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids,\
+                                        first_hb_transmit_expire_time)
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # CHECK
+    # Check first two CBSDs with a successful HB response
+    for response_num, resp in enumerate(response[:2]):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        grant_expire_time = datetime.strptime(resp['grantExpireTime'],\
```

Comment:
We don't expect grantExpireTime to be set in the HeartbeatResponse (except if grantRenew==True, which it does not in this case). This value should be obtained from the GrantResponse.